include/triton/Conversion/TritonGPUToLLVM/AsmFormat.h

```diff
@@ -0,0 +1,33 @@
+#ifndef TRITON_CONVERSION_TRITON_GPU_TO_LLVM_ASM_FORMAT_H_
+#define TRITON_CONVERSION_TRITON_GPU_TO_LLVM_ASM_FORMAT_H_
+
+#include "mlir/IR/Value.h"
+#include "triton/Dialect/Triton/IR/Dialect.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include <memory>
+#include <string>
+
+namespace mlir {
+class ConversionPatternRewriter;
+class Location;
+
+namespace triton {
+using llvm::StringRef;
+
+std::string strJoin(llvm::ArrayRef<std::string> strs,
+                    llvm::StringRef delimiter) {
+  std::string osStr;
+  llvm::raw_string_ostream os(osStr);
+  for (size_t i = 0; !strs.empty() && i < strs.size() - 1; ++i)
+    os << strs[i] << delimiter;
```

Comment:
The final string is not joined. Is this the right behavior? 
If yes, you may want to add a comment.

Reply:
we moved this existing function from PtxAsmFormat.cpp to a common header.

unfortunately, it is not clear to me whether this is a bug or expected behavior.

once we have more functionality added to the PR, then I think we can test it to verify.

thank you for pointing it out!